modules/tinymce/src/themes/silver/main/ts/api/Settings.ts

```diff
@@ -135,6 +134,22 @@ export interface ToolbarGroupSetting {
 
 const getToolbar = (editor: Editor): Array<string | ToolbarGroupSetting> | string | boolean => editor.getParam('toolbar', true);
 
+const getFilePickerCallback = (editor: Editor) => editor.getParam('file_picker_callback');
+
+const getFilePickerTypes = (editor: Editor) => editor.getParam('file_picker_types');
+
+const getFileBrowserCallbackTypes = (editor: Editor) => editor.getParam('file_browser_callback_types');
+
+const noTypeHeadUrls = (editor: Editor) => editor.getParam('typeahead_urls') === false;
+
+const getAnchorTop = (editor: Editor): string => editor.getParam('anchor_top', '#top', 'string');
+
+const getAnchorBottom = (editor: Editor): string => editor.getParam('anchor_bottom', '#bottom', 'string');
+
+const getFilePickerValidatorHandler = (editor: Editor) => editor.getParam('file_picker_validator_handler');
+
+const getFilePickerValidatorHandlerDepricatedVersion = (editor: Editor) => editor.getParam('filepicker_validator_handler');
```

Comment:
This doesn't need to be a setting function and can be merged into the `getFilePickerValidatorHandler` function. eg:
```
const getFilePickerValidatorHandler = (editor: Editor) => {
  const handler = editor.getParam('file_picker_validator_handler', undefined, 'function');
  if (handler === undefined) {
    // Fallback to legacy/deprecated setting
    return editor.getParam('filepicker_validator_handler', undefined, 'function');
  } else {
    return handler;
  }  
}
```

Then where it's being used, it can just do this instead:
```
export const getValidationHandler = (editor: Editor): Option<UrlValidationHandler> =>
  Option.from(Settings.getFilePickerValidatorHandler(editor));
```

Reply:
If we don't do the above, though I think we should as it keeps the fallback logic in one place, then there's a typo in the name.